examples/background-image.py

```diff
@@ -1,18 +1,37 @@
-from gi.repository import Caja, GObject, Gio
-
 SUPPORTED_FORMATS = 'image/jpeg', 'image/png'
 BACKGROUND_SCHEMA = 'org.mate.background'
-BACKGROUND_KEY = 'picture-uri'
+BACKGROUND_KEY = 'picture-filename'
+
+try:
+    # Python 3.
+    from urllib.parse import unquote, urlparse
+except:
+    # Python 2.
+    from urllib import unquote
+    from urlparse import urlparse
+
+from gi.repository import Caja, GObject, Gio
+
 
 class BackgroundImageExtension(GObject.GObject, Caja.MenuProvider):
     def __init__(self):
         self.bgsettings = Gio.Settings.new(BACKGROUND_SCHEMA)
     
+    def _filepath(self, file):
+        try:
+            file = file.get_uri()
+        except:
+            pass
+        (scheme, netloc, path, parameters, query, fragment) = urlparse(file)
+        if scheme and unquote(scheme) != 'file':
+            return None
+        return unquote(path)
+
     def menu_activate_cb(self, menu, file):
         if file.is_gone():
             return
         
-        self.bgsettings[BACKGROUND_KEY] = file.get_uri()
+        self.bgsettings[BACKGROUND_KEY] = self._filepath(file)
```

Comment:
Hmm... can we just use some other method from `GFile` to get the filename? It would keep the code simple as before.